bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/mapping/VolumioData.java

```diff
@@ -0,0 +1,360 @@
+
+        if (!artist.equals(this.artist)) {
+            this.artist = artist;
+            this.artistDirty = true;
+        } else {
+            this.artistDirty = false;
+        }
+    }
+
+    public PercentType getVolume() {
+        return new PercentType(volume);
+    }
+
+    public void setVolume(int volume) {
+        if (volume != this.volume) {
+            this.volume = volume;
+            this.volumeDirty = true;
+        } else {
+            this.volumeDirty = false;
+        }
+    }
+
+    public void setState(String state) {
+        if (!state.equals(this.state)) {
+            this.state = state;
+            this.stateDirty = true;
+        } else {
+            this.stateDirty = false;
+        }
+    }
+
+    public PlayPauseType getState() {
+        PlayPauseType playPauseStatus;
+
+        if ("play".equals(state)) {
+            playPauseStatus = PlayPauseType.PLAY;
+        } else {
+            playPauseStatus = PlayPauseType.PAUSE;
+        }
+
+        return playPauseStatus;
+    }
+
+    public void setTrackType(String trackType) {
+        if (!trackType.equals(this.trackType)) {
+            this.trackType = trackType;
+            this.trackTypeDirty = true;
+        } else {
+            this.trackTypeDirty = false;
+        }
+    }
+
+    public StringType getTrackType() {
+        return new StringType(trackType);
+    }
+
+    public void setPosition(String position) {
+        if (!position.equals(this.position)) {
+            this.position = position;
+            this.positionDirty = true;
+        } else {
+            this.positionDirty = false;
+        }
+    }
+
+    public void setCoverArt(@Nullable String coverArtUrl) {
+        if (coverArtUrl != null) {
+            if (!Objects.equals(coverArtUrl, this.coverArtUrl)) {
+                if (!coverArtUrl.startsWith("http")) {
+                    return;
+                }
+
+                try {
+                    URL url = new URL(coverArtUrl);
+                    URLConnection connection = url.openConnection();
+                    InputStream inStream = null;
+                    inStream = connection.getInputStream();
+                    coverArt = inputStreamToByte(inStream);
+                } catch (IOException ioe) {
+                    coverArt = null;
+                }
+                this.coverArtDirty = true;
+            } else {
+                this.coverArtDirty = false;
+            }
+        } else {
+            coverArt = null;
+        }
+    }
+
+    private byte @Nullable [] inputStreamToByte(InputStream is) {
+        byte @Nullable [] imgdata = null;
+        try {
+            ByteArrayOutputStream bytestream = new ByteArrayOutputStream();
+            int ch;
+
+            while ((ch = is.read()) != -1) {
+                bytestream.write(ch);
+            }
+            imgdata = bytestream.toByteArray();
+            bytestream.close();
+            return imgdata;
+        } catch (Exception e) {
```

Comment:
Use try-with-resources:

```suggestion
        try (ByteArrayOutputStream bytestream = new ByteArrayOutputStream()) {
            int ch;

            while ((ch = is.read()) != -1) {
                bytestream.write(ch);
            }
            imgdata = bytestream.toByteArray();
            return imgdata;
        } catch (Exception e) {
```

Reply:
solved